addon/ng2/models/config.ts

```diff
@@ -0,0 +1,123 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+
+export const CLI_CONFIG_FILE_NAME = 'angular-cli.json';
+
+
+export interface CliConfigJson {
+  routes?: { [name: string]: any },
+  packages?: { [name: string]: any }
+}
+
+
+function _findUp(name: string, from: string) {
+  let currentDir = from;
+  while (currentDir && currentDir != '/') {
+    const p = path.join(currentDir, name);
+    if (fs.existsSync(p)) {
+      return p;
+    }
+
+    currentDir = path.resolve(currentDir, '..');
+  }
+
+  return null;
+}
+
+
+export class CliConfig {
+  constructor(private _config: CliConfigJson = CliConfig.fromProject()) {}
+
+  save(path: string = CliConfig.configFilePath()) {
+    if (!path) {
+      throw new Error('Could not find config path.');
+    }
+
+    fs.writeFileSync(path, JSON.stringify(this._config, null, 2), { encoding: 'utf-8' });
+  }
+
+  set(jsonPath: string, value: any, force: boolean = false): boolean {
+    let { parent, name, remaining } = this._findParent(jsonPath);
+    while (force && remaining) {
+      if (remaining.indexOf('.') != -1) {
+        parent[name] = [];
```

Comment:
Please note that `get` is relatively easy to make, but `set` is significantly harder.